clusterman/autoscaler/pool_manager.py

```diff
@@ -546,12 +551,13 @@ def _prioritize_killable_nodes(self, killable_nodes: List[ClusterNodeMetadata])
 
         def sort_key(
             node_metadata: ClusterNodeMetadata,
-        ) -> Tuple[int, int, int, int, int, int]:
+        ) -> Tuple[int, int, int, int, int, int, int]:
             return (
                 0 if node_metadata.agent.is_frozen else 1,
                 0 if node_metadata.agent.state == AgentState.ORPHANED else 1,
-                0 if node_metadata.agent.state == AgentState.IDLE else 1,
                 0 if node_metadata.instance.is_stale else 1,
```

Comment:
i think we'll be able to get rid of this once we've gotten rid of our SFRs (which we only use for mesos stuff) - after that, I don't think any of the staleness code will be used since, afaict, it's only used when we're rolling spot fleets

(only mentioning this since the pr description mentions updating the priority for stale nodes)

Reply:
Gotcha. The main reason of the priority change was that I wanted the new sort component to sit in front of `AgentState.IDLE`, but `stale` was in the middle, and in my eyes something manually marked as "bad" should have definitely have higher priority than uptime or task count.